backend/open_webui/models/knowledge.py

```
@@ -174,61 +368,268 @@ def get_knowledge_bases_by_user_id(
             log.exception(e)
             return None
 
     def update_knowledge_data_by_id(
-        self, id: str, data: dict
+        self, id: str, data: dict, db: Optional[Session] = None
     ) -> Optional[KnowledgeModel]:
         try:
-            with get_db() as db:
-                knowledge = self.get_knowledge_by_id(id=id)
+            with get_db_context(db) as db:
+                knowledge = self.get_knowledge_by_id(id=id, db=db)
```

Ruff (0.14.14)

[error] Local variable `knowledge` is assigned to but never used

Remove assignment to unused variable `knowledge`

(F841)